### DISCUSSION TITLE
Booting into kernel panic on Lenovo T14 AMD gen 3 - Ryzen 6650U

### DISCUSSION THREAD

**Discussion by givtrah** [SpiralLinux/SpiralLinux-project#127] [Q&A]
I tried installing spirallinux, but I boot into a kernel panic, see attached images for both kernels.

Using spirallinux version: SpiralLinux_Plasma_11.220628_x86-64.iso

5.18.0-0.bpo.1 kernel safemode

![2022-09-18 14 48 47](https://user-images.githubusercontent.com/86203355/190903078-5bbc143e-b8e4-4c81-af14-5feffffbce65.jpg)

5.10.0-15 kernel safemode

![2022-09-18 14 52 59](https://user-images.githubusercontent.com/86203355/190903082-343ebd6e-d117-4286-8ddf-8a53a302fb54.jpg)

I have no idea what's going on.

**Comment by geckolinux** [comment]
Yikes, that's ugly. This is what I found by searching for "fatal exception in interrupt", also on a Lenovo laptop:
https://bbs.archlinux.org/viewtopic.php?pid=1965927#p1965927
So could you please try booting it with that additional kernel parameter `rcutree.use_softirq=0`  ?

**Reply by givtrah** [reply]
Unfortunately I still get a kernel panic, looks similar to the two I posted above. 

I have a functioning arch linux install on the laptop, so maybe it's something that's fixed in a newer kernel.

**Reply by geckolinux** [reply]
Yes it could be something with the specific kernel configuration that Debian uses.
By any chance could you try disabling SecureBoot to see if that makes any difference?
Also, as a last attempt, please try booting with `noacpi acpi=off` .

**Reply by givtrah** [reply]
SecureBoot is already off (Arch heh), noacpi acpi=off parameters resulted in a "BOOT FAILED!" 
![2022-09-19 06 18 57](https://user-images.githubusercontent.com/86203355/190949234-9e9a0213-c673-41b6-b86a-9f3767cc7f8d.jpg)

**Reply by givtrah** [reply]
I just tested, I can boot the weekly firmware testing debian image just fine, I'll try debian stable later.

Edit: Debian stable, firmware-11.5.0-amd6-netinst.iso also works.

**Reply by geckolinux** [reply]
Thanks a lot for the vanilla Debian tests, that helps to isolate the source of the problem. 

I imagine you checked the MD5sum for your downloaded ISO? And what method did you use to write the ISO to the USB?

If possible please try these as well:
- Possible candidates for the next SpiralLinux release: https://sourceforge.net/projects/spirallinux/files/staging/
- Debian Stable live + non-free firmware: https://cdimage.debian.org/images/unofficial/non-free/images-including-firmware/current-live/amd64/iso-hybrid/

**Reply by givtrah** [reply]
I successfully ran the spiral linux install on another computer earlier today.

I used Balena Etcher (windows..) to write the ISO. 

I have to admit I did not check the MD5sum yet as I was just testing things... I'll do that later but as it booted on another computer it should be fine. 

I'll check the candidate releases this evening and report.

Thanks for following up!

**Reply by geckolinux** [reply]
OK, thanks for the reply, assuming that it was the same USB key on both machines then we can discard that as the source of the problem.
Appreciate the additional testing!

**Reply by givtrah** [reply]
I managed to boot into the debian stable live cd and into the 5.10.0 kernel on the candidate spirallinux release.

However, in order to do so I had to delete most of the boot parameters which would cause a boot failure and then once the machine was fully booted I could switch TTY and startx as root.

Something is not right!

What do you need from me to fix this (note: This may be an upstream Debian problem in which case I have no idea how to proceed...)

**Reply by givtrah** [reply]
Final note: I got it installed, but only kernel 5.10.0 worked and with the caveat that booting didn't start SDDM, but I could switch TTY and run startx. 

Kernel 5.18.x and 5.19.x both refused to boot giving me stalling at amdgpu - which after some searching around seem to be due to some firmware missing from those kernels despite the package being installed.

**Reply by geckolinux** [reply]
Really appreciate the details and additional testing. I'm sorry I can't be of more help, as it is indeed an upstream Debian issue. That was confirmed by your test with the Debian Stable live image, which you say behaves the same as SpiralLinux if I understood correctly? So in that case I imagine it's probably due to the inclusion of some firmware or driver package that gets probed in the live system and causes a kernel fault on your hardware. The Debian installer loads a much more minimal image to bootstrap the system and only installs packages that are specific to your hardware, whereas the live image contains support for all possible hardware. You could definitely try filing a Debian bug report, but it might not be handled given that the live ISO even for vanilla Debian is labelled as "unofficial". Sorry again I couldn't be of more help.
